hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -1,11 +1,70 @@
 module HookContextExtension
+  # FIXME: remove when #23332 is released
+  def param_value(mod, name)
+    param(mod, name).value if param(mod, name)
+  end
+
   def success_file
     File.join(File.dirname(kafo.config.config_file), '.installed')
   end
 
   def new_install?
     !File.exist?(success_file)
   end
+
+  def fail_and_exit(message)
+    log_and_say :error, message
+    exit 1
+  end
+
+  def local_postgresql?
+    param_value('foreman', 'db_manage') || param_value('katello', 'candlepin_manage_db')
+  end
+
+  def log_and_say(level, message, do_say = true, do_log = true)
+    style = case level
+            when :error
+              'bad'
+            when :debug
+              'yellow'
+            else
+              level
+            end
+
+    # \ and ' characters could cause trouble in ERB, make sure to escape them
+    escaped_message = message.gsub('\\', '\\\\\\').gsub("'", %q{\\\'})
+    say "<%= color('#{escaped_message}', :#{style}) %>" if do_say
```

Comment:
How about directly calling it so we don't need to escape:
```suggestion
    say HighLine.color(message, style.to_sym) if do_say
```